taxi/tests/test_views.py

```diff
@@ -0,0 +1,64 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase, Client
+from django.urls import reverse
+
+from taxi.models import Manufacturer
+
+LITERARY_FORMAT_URL = reverse("taxi:manufacturer-list")
+
+
+class PublicManufacturerTests(TestCase):
+
+    def test_login_required(self):
+        res = self.client.get(LITERARY_FORMAT_URL)
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateManufacturerTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer(self):
+        Manufacturer.objects.create(name="Tesla")
+        Manufacturer.objects.create(name="Mercedes-Benz")
+
+        response = self.client.get(LITERARY_FORMAT_URL)
+
+        manufacturer = Manufacturer.objects.all()
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturer)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+
+class PrivateDriverTests(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            "test",
+            "password12345"
+        )
+        self.client.force_login(self.user)
+
+    def test_create_driver(self):
+        form_data = {
+            "username": "new_user",
+            "password1": "user123test",
+            "password2": "user123test",
+            "first_name": "Test first",
+            "last_name": "Test last",
+            "license_number": "QWE12345"
+        }
+
+        self.client.post(reverse("taxi:driver-create"), data=form_data)
+        new_user = get_user_model().objects.get(username=form_data["username"])
+
+        self.assertEqual(new_user.first_name, form_data["first_name"])
+        self.assertEqual(new_user.last_name, form_data["last_name"])
+        self.assertEqual(new_user.license_number, form_data["license_number"])
```

Comment:
There should be more test, like PublicDriverTests, tests for Car view

Reply:
added